### PR TITLE
DE - Refactor remaining classes to use generic handlers

### PR DESCRIPTION
**Pull request by alexandra-bucur**
Signed-off-by: Alexandra Bucur <alexandra-nona.bucur@ing.com>

A few Data Engine handlers still used the repository handler instead of the generic handlers - OpenMetadataAPIGenericHandler, so a refactoring was needed. 

## Description

- refactored DE handlers
- removed unused methods from RepositoryHandler

## Testing

JUnits, manual, FVT

### CODE REVIEW

File: open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineCommonHandler.java [hunk 1]
```diff
@@ -207,15 +155,19 @@ public Optional<EntityDetail> findEntity(String userId, String qualifiedName, St
      * @throws UserNotAuthorizedException user not authorized to issue this request.
      * @throws PropertyServerException    problem retrieving the entity.
      */
-    public Optional<EntityDetail> getEntityDetails(String userId, String entityDetailGUID, String entityTypeName) throws InvalidParameterException,
-                                                                                                                         PropertyServerException,
-                                                                                                                         UserNotAuthorizedException {
+    public Optional<EntityDetail> getEntityDetails(String userId, String entityDetailGUID, String entityTypeGUID,
```
Comment (lcpopa) [inline open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineCommonHandler.java:158]: Git warning: The parameter entityTypeGUID is unused

File: open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineCommonHandler.java [hunk 2]
```diff
@@ -248,20 +201,27 @@ protected void upsertExternalRelationship(String userId, String firstGUID, Strin
...
-        Optional<Relationship> relationship = findRelationship(userId, firstGUID, secondGUID, firstEntityTypeName, relationshipTypeName);
+        Optional<Relationship> relationship = findRelationship(userId, firstGUID, secondGUID, firstEntityTypeName,
+                secondEntityTypeName, relationshipTypeName);
         if (relationship.isEmpty()) {
 
             TypeDef relationshipTypeDef = repositoryHelper.getTypeDefByName(userId, relationshipTypeName);
-            repositoryHandler.createExternalRelationship(userId, relationshipTypeDef.getGUID(), externalSourceGUID, externalSourceName,
-                    firstGUID, secondGUID, relationshipProperties, methodName);
+
+            String relationshipGUID = genericHandler.linkElementToElement(userId, externalSourceGUID, externalSourceName, firstGUID,
```
Comment (lcpopa) [inline open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineCommonHandler.java:210]: relationshipGUID is never used also. Not assigning the returned value of linkElementToElement should fix it

**Review by lcpopa — CHANGES_REQUESTED** [review]

**Comment by planetf1** [comment]
I noticed the DCO check is 'stuck'. I suspect it will be ok once the branch is updated.

**Comment by alexandra-bucur** [comment]
@planetf1 thank you! it worked at the rerun.

File: open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineCommonHandler.java [hunk 3]
```diff
@@ -28,20 +27,23 @@
...
 import java.util.stream.Collectors;
 
+import static org.odpi.openmetadata.commonservices.generichandlers.OpenMetadataAPIMapper.GUID_PROPERTY_NAME;
+
 /**
  * DataEngineCommonHandler manages objects from the property server. It runs server-side in the DataEngine OMAS
  * and creates port entities with wire relationships through the OMRSRepositoryConnector.
  */
 public class DataEngineCommonHandler {
     private final String serviceName;
```
[the remark this answers is not in the document]
Comment (alexandra-bucur) [reply open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineCommonHandler.java:44]: Common pattern to use the service name information in access services. This needs to be refactored in order to be used. Won't fix this for the moment.

**Review by lcpopa — APPROVED** [review]

**Pull request merged**